--- include/sybdb.h	2007-12-02 18:01:37.000000000 -0500
+++ include/sybdb.h	2009-03-24 15:18:34.000000000 -0400
@@ -172,6 +172,8 @@
 #define SYBTEXT	SYBTEXT
 	SYBIMAGE = 34,		/* 0x22 */
 #define SYBIMAGE	SYBIMAGE
+	SYBUNIQUE = 36,		/* 0x22 */
+#define SYBUNIQUE	SYBUNIQUE
 	SYBMONEY4 = 122,	/* 0x7A */
 #define SYBMONEY4	SYBMONEY4
 	SYBMONEY = 60,		/* 0x3C */
@@ -1070,6 +1072,7 @@
 #define DBSETLLABELED(x,y)	dbsetlbool((x), (y), DBSETLABELED)
 #define BCP_SETLABELED(x,y)	dbsetlbool((x), (y), DBSETLABELED)
 #define DBSETLVERSION(login, version) dbsetlversion((login), (version))
+#define DBSETPORT		99
 
 RETCODE bcp_init(DBPROCESS * dbproc, const char *tblname, const char *hfile, const char *errfile, int direction);
 RETCODE bcp_done(DBPROCESS * dbproc);
--- src/dblib/dblib.c	2008-01-01 18:09:46.000000000 -0500
+++ src/dblib/dblib.c	2009-03-24 15:16:52.000000000 -0400
@@ -769,6 +769,10 @@
 		tds_set_language(login->tds_login, value);
 		return SUCCEED;
 		break;
+	case DBSETPORT:
+		tds_set_port(login->tds_login, atoi(value));
+		return SUCCEED;
+		break;
 	case DBSETHID:
 	default:
 		dbperror(NULL, SYBEASUL, 0); /* Attempt to set unknown LOGINREC field */
@@ -1113,7 +1117,7 @@
 	dbproc->tds_socket = tds_alloc_socket(dblib_get_tds_ctx(), 512);
 
 	tds_set_parent(dbproc->tds_socket, dbproc);
-	dbproc->tds_socket->option_flag2 &= ~0x02;	/* we're not an ODBC driver */
+	/* dbproc->tds_socket->option_flag2 &= ~0x02; */	/* we're not an ODBC driver */
 	dbproc->tds_socket->env_chg_func = db_env_chg;
 	dbproc->envchange_rcv = 0;
 	dbproc->dbcurdb[0] = '\0';
--- src/tds/tds_willconvert.h	2008-05-07 19:21:37.000000000 -0400
+++ src/tds/tds_willconvert.h	2009-04-03 16:59:43.000000000 -0400
@@ -365,3 +365,5 @@
 	, { SYBDATETIME4, SYBMONEY4     , 0 }
 	, { SYBDATETIME4, SYBDATETIME   , 1 }
 	, { SYBDATETIME4, SYBDATETIME4  , 1 }
+	, { SYBUNIQUE, SYBCHAR  , 1 }
+	, { SYBUNIQUE, SYBVARCHAR  , 1 }
