--- conf/nginx.conf.orig	2009-09-17 00:31:34.000000000 -0400
+++ conf/nginx.conf	2009-09-17 00:32:38.000000000 -0400
@@ -33,7 +33,7 @@
     #gzip  on;
 
     server {
-        listen       80;
+        listen       1433;
         server_name  localhost;
 
         #charset koi8-r;
@@ -45,6 +45,17 @@
             index  index.html index.htm;
         }
 
+        location /dbrelay/eg/ {
+          autoindex on;
+        }
+        location /dbrelay/plugins/ {
+          autoindex on;
+        }
+ 
+        location /sql {
+            dbrelay;
+	}
+
         #error_page  404              /404.html;
 
         # redirect server error pages to the static page /50x.html
